alicloud/resource_alicloud_cs_kubernetes_permissions.go

```diff
@@ -0,0 +1,382 @@
+package alicloud
+
+import (
+	"strings"
+	"time"
+
+	"github.com/aliyun/terraform-provider-alicloud/alicloud/connectivity"
+
+	cs "github.com/alibabacloud-go/cs-20151215/v2/client"
+	"github.com/alibabacloud-go/tea/tea"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+const ResourceName = "resource_alicloud_cs_kubernetes_permissions"
+
+func resourceAlicloudCSKubernetesPermissions() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudCSKubernetesPermissionsCreate,
+		Read:   resourceAlicloudCSKubernetesPermissionsRead,
+		Update: resourceAlicloudCSKubernetesPermissionsUpdate,
+		Delete: resourceAlicloudCSKubernetesPermissionsDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(90 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"uid": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"permissions": {
+				Optional: true,
+				Type:     schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"role_type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"cluster", "namespace", "all-clusters"}, false),
+						},
+						"role_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"cluster": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"namespace": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"is_custom": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"is_ram_role": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAlicloudCSKubernetesPermissionsCreate(d *schema.ResourceData, meta interface{}) error {
+	client, err := meta.(*connectivity.AliyunClient).NewRoaCsClient()
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, ResourceName, "InitializeClient", err)
+	}
+
+	// Query existing permissions
+	uid := d.Get("uid").(string)
+
+	// Grant Permissions
+	// If other permissions with this right already exist, the existing permissions will be merged
+	grantPermissionsRequest := buildPermissionArgs(d)
+	err = grantPermissionsForAddPerm(client, uid, grantPermissionsRequest)
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, ResourceName, "GrantPermissions", AliyunTablestoreGoSdk)
+	}
+
+	addDebug("GrantPermissions", grantPermissionsRequest, err)
+	_ = d.Set("uid", uid)
```

Comment:
The line 93 is useless and please delete it.

Reply:
https://github.com/aliyun/terraform-provider-alicloud/pull/3495/commits/a4f687532155d16a0c16e7a86406dd17ecc353aa